src/statics.rs

```diff
@@ -80,29 +87,98 @@ pub(crate) static EVENT_METADATA: once_cell::sync::Lazy<
     }
 }
 
 impl core::cmp::Eq for crate::_details::ParsedEventMetadata {}
 
 impl core::cmp::PartialOrd for crate::_details::ParsedEventMetadata {
     fn partial_cmp(&self, other: &Self) -> Option<cmp::Ordering> {
-        self.identity_hash.partial_cmp(&other.identity_hash)
+        match self.identity_hash.partial_cmp(&other.identity_hash) {
+            None => None,
+            Some(cmp::Ordering::Equal) => {
+                match self.meta.identity == other.meta.identity {
+                    true => Some(cmp::Ordering::Equal),
+                    false => {
+                        // We need to do *something* to sort two different callsites that hit a hash collision.
+                        // TODO: This only works when comparing the static entries generated by the logging macros.
+                        let lhs = &self.meta.identity as *const tracing_core::callsite::Identifier as usize;
+                        let rhs = &other.meta.identity as *const tracing_core::callsite::Identifier as usize;
+                        lhs.partial_cmp(&rhs)
+                    }
+                }
+            },
+            Some(x) => Some(x)
+        }
     }
 }
 
 impl core::cmp::Ord for crate::_details::ParsedEventMetadata {
     fn cmp(&self, other: &Self) -> std::cmp::Ordering {
-        self.identity_hash.cmp(&other.identity_hash)
+        match self.identity_hash.cmp(&other.identity_hash) {
+            cmp::Ordering::Equal => {
+                match self.meta.identity == other.meta.identity {
+                    true => cmp::Ordering::Equal,
+                    false => {
+                        // We need to do *something* to sort two different callsites that hit a hash collision.
+                        // TODO: This only works when comparing the static entries generated by the logging macros.
+                        let lhs = &self.meta.identity as *const tracing_core::callsite::Identifier as usize;
+                        let rhs = &other.meta.identity as *const tracing_core::callsite::Identifier as usize;
+                        lhs.cmp(&rhs)
+                    }
+                }
+            },
+            x => x
+        }
     }
 }
 
-pub(crate) fn get_event_metadata(id: &tracing::callsite::Identifier) -> Option<&'static crate::_details::ParsedEventMetadata> {
-    let hash = BH.hash_one(id);
-    let etw_meta = EVENT_METADATA.binary_search_by_key(&hash, |m| { m.identity_hash });
-    match etw_meta {
-        Ok(idx) => Some(&EVENT_METADATA[idx]),
-        _ => None
+pub(crate) fn get_event_metadata(id: &tracing::callsite::Identifier) -> Option<&'static crate::_details::EventMetadata> {
+    let bh = FnvHasher::default();
+    let identity_hash = bh.hash_one(id);
+    let idx = EVENT_METADATA.partition_point(|other| other.identity_hash > identity_hash);
+    let mut cur = idx;
+    while cur <EVENT_METADATA.len() {
+        let meta = &EVENT_METADATA[cur];
+        if meta.identity_hash == identity_hash {

```

Comment:
Style: If you test for `meta.identity_hash != identity_hash` then you can reduce the level of nesting. I also think it reads more clearly "if the hashes don't match, break" rather than "if the hashes match, do stuff and more stuff and more stuff, else break".

```rust
if meta.identity_hash != identity_hash {
    break;
}
if (meta.meta.identity == *id {
    return Some(meta.meta);
}
cur += 1;
```

Optionally you can make it an if-else chain, but that's up to your personal preference.